File: scratch/patch_imports.py
import glob
import os
import re

files = glob.glob('src/pages/**/*.tsx', recursive=True)
files.extend(glob.glob('../pouch-eco-website/src/app/**/*.tsx', recursive=True))

for fpath in files:
    with open(fpath, 'r') as f:
        content = f.read()
    
    original = content
    
    # 1. Fix SEOPageLayout import
    # Find depth
    # e.g. src/pages/packaging/CustomBoxesPage.tsx -> 2 slashes after src/pages
    parts = fpath.split('/')
    if 'pages' in parts:
        depth = len(parts) - parts.index('pages') - 1
        prefix = '../' * depth
        content = re.sub(r"import SEOPageLayout from '.*?/components/seo/SEOPageLayout';", f"import SEOPageLayout from '{prefix}components/SEOPageLayout';", content)
    
    # 2. Fix PouchLayout import
    # If in src/pages/pouch/packaging/PouchCustomBoxesPage.tsx
    content = content.replace("import PouchLayout from '../PouchLayout';", "import PouchLayout from '../../pouch/PouchLayout';")
    
    # If in src/pages/solutions/PackagingLineAutomationPage.tsx
    content = content.replace("import PouchLayout from '../pouch/PouchLayout';", "import PouchLayout from '../../components/pouch/PouchLayout';")
    
    # 3. Fix domain import
    content = content.replace("import { isPouch } from '../../utils/domainHelper';", "import { isPouch } from '../../utils/domain';")

    if content != original:
        with open(fpath, 'w') as f:
            f.write(content)
        print('Patched', fpath)

File: scratch/patch_imports.js
const fs = require('fs');
const glob = require('glob');

const files = glob.sync('src/pages/**/*.tsx');

files.forEach(file => {
  let content = fs.readFileSync(file, 'utf8');
  let original = content;

  content = content.replace(/import SEOPageLayout from '.*?\/components\/seo\/SEOPageLayout';/g, (match) => {
    // Determine the right relative path based on file depth
    const depth = file.split('/').length - 2; // src/pages is depth 2, minus 2 for src/components
    const prefix = '../'.repeat(depth);
    return `import SEOPageLayout from '${prefix}components/SEOPageLayout';`;
  });
  
  content = content.replace(/import PouchLayout from '\.\.\/PouchLayout';/g, (match) => {
    // For files in src/pages/pouch/packaging/PouchCustomBoxesPage.tsx
    return `import PouchLayout from '../../pouch/PouchLayout';`;
  });
  
  content = content.replace(/import PouchLayout from '\.\.\/pouch\/PouchLayout';/g, (match) => {
    // For files in src/pages/solutions/PackagingLineAutomationPage.tsx
    return `import PouchLayout from '../../components/pouch/PouchLayout';`;
  });

  content = content.replace(/import \{ isPouch \} from '\.\.\/\.\.\/utils\/domainHelper';/g, "import { isPouch } from '../../utils/domain';");

  if (content !== original) {
    fs.writeFileSync(file, content, 'utf8');
    console.log('Patched', file);
  }
});
